packages/smooth_app/lib/background/background_task_manager.dart

```diff
@@ -136,32 +147,49 @@ class BackgroundTaskManager {
   /// we can remove the failed task from the list: it would have been
   /// overwritten anyway.
   Future<void> run() async {
-    if (!await _canStartNow()) {
+    final int? now = _canStartNow();
+    if (now == null) {
       return;
     }
-    final List<BackgroundTask> tasks = await _getAllTasks();
-    for (final BackgroundTask task in tasks) {
-      await task.recover(localDatabase);
-    }
-    for (final BackgroundTask task in tasks) {
-      final String taskId = task.uniqueId;
-      try {
-        await _setTaskErrorStatus(taskId, taskStatusStarted);
-        await task.execute(localDatabase);
-        await _finishTask(taskId, success: true);
-      } catch (e) {
-        // Most likely, no internet, no reason to go on.
-        if (e.toString().startsWith('Failed host lookup: ')) {
-          await _setTaskErrorStatus(taskId, taskStatusNoInternet);
-          await _justFinished();
-          return;
+    _running = true;
+
+    ///
+    /// Will also set the "latest start timestamp".
+    /// With this, we can detect a run that went wrong.
```

Comment:
Can we perhaps instrument those anomalies in Matomo ?

Reply:
Sure, in a new issue.

Reply:
👍